components/layout/DocsLayout.js

```diff
@@ -17,13 +17,24 @@ import { SearchButton, DOCS_INDEX_NAME } from '../AlgoliaSearch';
 import IconLoupe from '../icons/Loupe';
 import { getAllPosts } from '../../lib/api'
 import Link from 'next/link'
+import editOptions from '../../config/edit-page-config.json'
+
 
 function generateEditLink(post) {
-  if (post.slug.includes('/specifications/')) {
-    return <a target="_blank" rel="noopener noreferrer" href={`https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md`} className="ml-1 underline">Edit this page on GitHub</a>
-  } 
-  return <a target="_blank" rel="noopener noreferrer" href={`https://github.com/asyncapi/website/blob/master/pages${post.isIndex ? post.slug + '/index' : post.slug}.md`} className="ml-1 underline">Edit this page on GitHub</a>
-}
+
+  var last=post.id.substring(post.id.lastIndexOf("/") + 1);
+  var target=editOptions.find(edit=>{ return post.slug.includes(edit.value)});
+  
+  if(target.value==""){
+    return <a target="_blank" rel="noopener noreferrer" href={`${target.href}${post.isIndex ? post.slug + '/index' : post.slug}.md`} className="ml-1 underline">Edit this page on GitHub</a>
+
+  }
+  if (target.value=="reference/specification/") last=""
```

Comment:
`reference/specification/` we still have hardcode in code, so config is not totally independent.

you can do something like this
pseudo code incoming:
```
const editHref = target.href;
//in case href points to a file, then edit link should always point to that file and not be dynamic
const hrefList = editHref.split('/');
const lastListElement = hrefList[hrefList.length - 1].split('.');
const isHrefToFile = lastListElement.length > 1;
```

so instead of 
```
if (target.value=="reference/specification/") last=""
```
you could do
```
if (isHrefToFile) last = ""
```